### ISSUE TITLE
Cached producer retains settings

### ISSUE THREAD

**Issue by TheCycoONE** [messaginghub/pooled-jms#40]
I'm not sure if there's interest in changing this, but it caused issues for an application I worked on.

Normally when you create a producer from a session it's a clean slate, priority 4, no delivery delay, etc., however when adding pooled-jms to an existing application we noticed that messages that shouldn't have had a delay at all were being scheduled.

I would expect pooled-jms to track what settings were changed and reset them either on close or create.

**Comment by tabish121** [comment]
Can you provide a test to show when this happens, the current code always has measures in place to avoid setting these values on the producer and does reset the delivery delay after each send.

**Comment by TheCycoONE** [comment]
https://github.com/messaginghub/pooled-jms/blob/main/pooled-jms/src/main/java/org/messaginghub/pooled/jms/JmsPoolMessageProducer.java#L306-L311

Replication is trivial:
```
import jakarta.jms.*;
import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory;
import org.messaginghub.pooled.jms.JmsPoolConnectionFactory;

import java.time.Duration;

public class Application {
    public static void main(String[] args) throws Exception {
        ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory("tcp://localhost:2412");
        connectionFactory.setUser("artemis");
        connectionFactory.setPassword("artemis");
        JmsPoolConnectionFactory jpcf = new JmsPoolConnectionFactory();
        jpcf.setConnectionFactory(connectionFactory);
        jpcf.setMaxConnections(1);
        jpcf.setMaxSessionsPerConnection(1);
        Application.send(jpcf, true);
        Application.send(jpcf, false);
    }

    private static void send(ConnectionFactory cf, boolean withDelay) {
        try (Connection connection = cf.createConnection()) {
            connection.start();
            try (Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE)) {
                Queue queue = session.createQueue("Test");
                MessageProducer producer = session.createProducer(queue);
                TextMessage message = session.createTextMessage("This is a test " + (withDelay ? "with delay" : "without delay"));
                if (withDelay) {
                    producer.setDeliveryDelay(Duration.ofHours(1).toMillis());
                }
                producer.send(message);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
```

```
plugins {
    id 'java'
}

group = 'com.raveu'
version = '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {
    implementation 'org.apache.activemq:artemis-jakarta-server:2.37.0'
    implementation 'org.messaginghub:pooled-jms:3.1.6'
}
```

**Comment by tabish121** [comment]
Fixed the underlying issue and released a new 3.1.7 version with this change.
